### PR TITLE
fix for create env on prow and add filtering to jx get build log on tekton

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

File: pkg/jx/cmd/get_build_logs.go [hunk 1]
```diff
@@ -502,7 +502,7 @@ func (o *GetBuildLogsOptions) loadPipelines(kubeClient kubernetes.Interface, tek
 			log.Warnf("Error creating PipelineRunInfo for PipelineRun %s: %s\n", pr.Name, err)
 			return names, defaultName, buildMap, pipelineMap, err
 		}
-		if pri != nil {
+		if pri != nil && o.BuildFilter.BuildMatches(pri.ToBuildPodInfo()) {
```
Comment (abayer) [inline pkg/jx/cmd/get_build_logs.go]: I think we'll eventually want to change `BuildFilter` to be a `PipelineRunInfoFilter`, but it does seem easier to just keep it compatible for now.

Comment (jstrachan) [reply pkg/jx/cmd/get_build_logs.go]: yeah - or we could have a filter interface which we support on both `BuildRunInfo` and `PipeilneRunInfo` that both take the filter parmeters and do their thing

**Pull request merged**